aries_cloudagent/vc/data_integrity/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,201 @@
+        proof_bytes = await self.proof_serialization(hash_data, options)
+
+        proof["proofValue"] = multibase.encode(proof_bytes, "base58btc")
+
+        return proof
+
+    def proof_configuration(self, options: dict):
+        """Proof configuration algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#proof-configuration-eddsa-jcs-2022.
+        """
+        proof_config = options.copy()
+
+        assert (
+            proof_config["type"] == "DataIntegrityProof"
+        ), 'Expected proof.type to be "DataIntegrityProof'
+        assert (
+            proof_config["cryptosuite"] == "eddsa-jcs-2022"
+        ), 'Expected proof.cryptosuite to be "eddsa-jcs-2022'
+
+        if "created" in proof_config:
+            # TODO assert proper [XMLSCHEMA11-2] dateTimeStamp string
+            assert proof_config[
+                "created"
+            ], "Expected proof.created to be a [XMLSCHEMA11-2] dateTimeStamp string."
+
+        if "expires" in proof_config:
+            # TODO assert proper [XMLSCHEMA11-2] dateTimeStamp string
+            assert proof_config[
+                "expires"
+            ], "Expected proof.expires to be a [XMLSCHEMA11-2] dateTimeStamp string."
+
+        return self._canonicalize(proof_config)
+
+    def transformation(self, unsecured_document: dict, options: dict):
+        """Transformation algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#transformation-eddsa-jcs-2022.
+        """
+        assert (
+            options["type"] == "DataIntegrityProof"
+        ), "Expected proof.type to be `DataIntegrityProof`"
+        assert (
+            options["cryptosuite"] == "eddsa-jcs-2022"
+        ), "Expected proof.cryptosuite to be `eddsa-jcs-2022`"
+
+        return self._canonicalize(unsecured_document)
+
+    def hashing(self, transformed_document, canonical_proof_config):
+        """Hashing algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#hashing-eddsa-jcs-2022.
+        """
+        return (
+            sha256(canonical_proof_config).digest()
+            + sha256(transformed_document).digest()
+        )
+
+    async def proof_serialization(self, hash_data: bytes, options: dict):
+        """Proof Serialization Algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#proof-serialization-eddsa-jcs-2022.
+        """
+        # If the verification method is a did:key: URI,
+        # we derive the signing key from a multikey value
+        if options["verificationMethod"].startswith("did:key:"):
+            multikey = options["verificationMethod"].split("#")[-1]
+            key_info = await self.key_manager.from_multikey(multikey)
+
+        # Otherwise we derive the signing key from a kid
+        else:
+            key_info = await self.key_manager.from_kid(options["verificationMethod"])
+
+        return await self.wallet.sign_message(
+            message=hash_data,
+            from_verkey=self.key_manager._multikey_to_verkey(key_info["multikey"]),
+        )
```

Comment:
I think we should consider updating the wallet interface to accept a multikey. Also a bit odd to call an underscore method outside of the key manager class.